### PR TITLE
[WIP] Change UQ parameter parsing/writing to work with new `ParameterDistribution` API

### PR DESCRIPTION
**Pull request by bielim**
This PR updates all UQ parameter parsing and writing functionality to work with the new `ParameterDistribution` API. 

- [x] read UQ parameters
- [x] write UQ parameters
- [x] update tests
- [x] add regularization flags
- [x] clean up and review

### CODE REVIEW

**Review by odunbar — APPROVED** [review]
Just some very small comments.

File: src/file_parsing_uq.jl [hunk 1]
```diff
@@ -682,8 +438,69 @@ Args:
...
+
+"""
+get_regularization(param_dict, names)
+
+Returns the regularization information for an array of parameters
+
+Args:
+`param_dict` - nested dictionary that has parameter names as keys and the
+               corresponding dictionary of parameter information as values
+`name` - parameter name
```
[the remark this answers is not in the document]
Comment (bielim) [reply src/file_parsing_uq.jl]: Yes, thanks.

File: src/file_parsing_uq.jl [hunk 2]
```diff
@@ -682,8 +438,69 @@ Args:
...
+
+Returns an arary of tuples (<regularization_type>, <regularization_value>), with
+the ith tuple corresponding to the parameter `names[i]`. The regularization
+type is either "L1" or "L2", and the regularization value is a float.
+Returns (nothing, nothing) for parameters that have no regularization
+information.
+"""
+function get_regularization(param_dict::Dict, names::Array{String, 1})
```
[the remark this answers is not in the document]
Comment (bielim) [reply src/file_parsing_uq.jl]: Corrected.

File: src/file_parsing_uq.jl [hunk 3]
```diff
@@ -32,119 +33,80 @@ end
...
+
+Args:
+`param_dict` - nested dictionary that has parameter names as keys and the
+               corresponding dictionary of parameter information (in particular,
+               the parameters' prior distributions and constraints) as values
+`names` - array of parameter names
+
+Returns a `ParameterDistribution` 
+"""
+function get_parameter_distribution(param_dict::Dict, names::Array{String, 1})
```
[the remark this answers is not in the document]
Comment (bielim) [reply src/file_parsing_uq.jl]: Corrected.

**Pull request merged**